kernel/src/process.rs

```diff
@@ -2075,6 +2075,28 @@ impl<C: 'static + Chip> Process<'_, C> {
         self.allow_high_water_mark
             .set(self.original_allow_high_water_mark);
 
+        // Reset the MPU configuration based on the original app_break, which is
+        // likely different than what the app break was when the app crashed.
+        let mpu_config_res = self.mpu_config.map_or(Err(()), |mut config| {
+            self.chip.mpu().update_app_memory_region(
+                self.app_break.get(),
+                self.kernel_memory_break.get(),
+                mpu::Permissions::ReadWriteOnly,
+                &mut config,
```

Comment:
This appears to still use the MPU config of the failed process, which is not guaranteed to be the same as the MPU config of a newly created process, as I understand it (`allocate_region()`, for example, modifies the config).

Should we be replacing the config with the `Default` before this?